sdks/python/apache_beam/metrics/metric.py

```diff
@@ -0,0 +1,161 @@
+# limitations under the License.
+#
+
+"""
+User-facing classes for Metrics API.
+
+The classes in this file allow users to define and use metrics to be collected
+and displayed as part of their pipeline execution.
+
+- Metrics - This class lets pipeline and transform writers create and access
+    metric objects such as counters, distributions, etc.
+"""
+import inspect
+
+from apache_beam.metrics.base import Counter, Distribution
+from apache_beam.metrics.base import MetricName
+from apache_beam.metrics.internal import MetricsEnvironment
+
+
+class Metrics(object):
+  """ Lets users create/access metric objects during pipeline execution.
+  """
+  @staticmethod
+  def get_namespace(namespace):
+    if inspect.isclass(namespace):
+      return '{}.{}'.format(namespace.__module__, namespace.__name__)
+    elif isinstance(namespace, str):
+      return namespace
+    else:
+      raise ValueError('Unknown namespace type')
+
+  @staticmethod
+  def counter(namespace, name):
+    """ Obtains or creates a Counter metric.
+
+    Args:
+      namespace - A class or string that gives the namespace to a metric
+      name - A string that gives a unique name to a metric
+
+    Returns:
+      An object of Counter class.
+    """
+    namespace = Metrics.get_namespace(namespace)
+    return Metrics.DelegatingCounter(MetricName(namespace, name))
+
+  @staticmethod
+  def distribution(namespace, name):
+    """ Obtains or creates a Distribution metric.
+
+    Args:
+      namespace - A class or string that gives the namespace to a metric
+      name - A string that gives a unique name to a metric
+
+    Returns:
+      An object of Distribution class.
+    """
+    namespace = Metrics.get_namespace(namespace)
+    return Metrics.DelegatingDistribution(MetricName(namespace, name))
+
+  class DelegatingCounter(Counter):
+    def __init__(self, metric_name):
+      self.metric_name = metric_name
+
+    def inc(self, n=1):
+      container = MetricsEnvironment.current_container()
+      if container is not None:
+        container.get_Counter(self.metric_name).inc(n)
+
+  class DelegatingDistribution(Distribution):
+    def __init__(self, metric_name):
+      self.metric_name = metric_name
+
+    def update(self, value):
+      container = MetricsEnvironment.current_container()
+      if container is not None:
+        container.get_Distribution(self.metric_name).update(value)
+
+
+class MetricResults(object):
+  @staticmethod
+  def matches(filter, metric_key):
+    if filter is None:
+      return True
+
+    if (metric_key.step in filter.steps and
+        metric_key.metric.namespace in filter.namespaces and
+        metric_key.metric.name in filter.names):
+      return True
+    else:
+      return False
+
+  def query(self, filter):
+    raise NotImplementedError
+
+
+class MetricsFilter(object):
+  """ Simple object to filter metrics results.
+
+  If filters by matching a result's step-namespace-name with three internal
```

Comment:
Is it worth mentioning that this doesn't include anything executable (such as a filter function) because the intention is that this could be used to construct the arguments to an RPC request?